{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phenopacket store statistics\n",
    "\n",
    "This notebook performs quality assessment and calculate descriptive statistics about a phenopacket-store release.\n",
    "\n",
    "Note: \n",
    "\n",
    "We recommend installing Phenopacket Store Toolkit into the notebook kernel:\n",
    "\n",
    "```shell\n",
    "python3 -m pip install phenopacket-store-toolkit[release]\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The input file is the ZIP file that is or will be added to each release.\n",
    "\n",
    "The ZIP file can be generated by running:\n",
    "\n",
    "```shell\n",
    "python3 -m ppktstore package --notebook-dir notebooks --release-tag 0.1.19 --output all_phenopackets\n",
    "```\n",
    "\n",
    "assuming `phenopacket-store-toolkit` was installed into the active environment, and `notebooks` points to the folder with Phenopacket store notebook directory.\n",
    "\n",
    "Note, we need to update the `--release-tag` option periodically."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "import zipfile\n",
    "from ppktstore.model import PhenopacketStore\n",
    "\n",
    "release_tag = '0.1.19'\n",
    "\n",
    "input_zip = \"/Users/leonardo/git/malco/in_multlingual_nov24/prompts/used_ppkts/\"\n",
    "#with zipfile.ZipFile() as zf:\n",
    "store = PhenopacketStore.from_notebook_dir(input_zip, \"jsons/\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can summarize statistics of the individuals described in the phenopackets, their phenotypic features, diseases, and genotypes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ppktstore.release.stats import PPKtStoreStats\n",
    "\n",
    "stats = PPKtStoreStats(store)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The summary "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>cohort</th>\n",
       "      <th>disease_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>gene</th>\n",
       "      <th>allele_1</th>\n",
       "      <th>allele_2</th>\n",
       "      <th>PMID</th>\n",
       "      <th>filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2210</th>\n",
       "      <td>Family 1 proband</td>\n",
       "      <td>singlecohort</td>\n",
       "      <td>OMIM:148600</td>\n",
       "      <td>Keratoderma, palmoplantar, punctate type IA</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>NM_024666.5:c.505_506dup</td>\n",
       "      <td></td>\n",
       "      <td>PMID:28239884</td>\n",
       "      <td>singlecohort/jsons/PMID_28239884_Family1proband.json</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3458</th>\n",
       "      <td>Family 3 proband</td>\n",
       "      <td>singlecohort</td>\n",
       "      <td>OMIM:148600</td>\n",
       "      <td>Keratoderma, palmoplantar, punctate type IA</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>NM_024666.5:c.870+1G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>PMID:28239884</td>\n",
       "      <td>singlecohort/jsons/PMID_28239884_Family3proband.json</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>Family 2 proband</td>\n",
       "      <td>singlecohort</td>\n",
       "      <td>OMIM:148600</td>\n",
       "      <td>Keratoderma, palmoplantar, punctate type IA</td>\n",
       "      <td>AAGAB</td>\n",
       "      <td>NM_024666.5:c.473del</td>\n",
       "      <td></td>\n",
       "      <td>PMID:28239884</td>\n",
       "      <td>singlecohort/jsons/PMID_28239884_Family2proband.json</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>II.2</td>\n",
       "      <td>singlecohort</td>\n",
       "      <td>OMIM:601718</td>\n",
       "      <td>Retinitis pigmentosa 19</td>\n",
       "      <td>ABCA4</td>\n",
       "      <td>NM_000350.3:c.1938-1G&gt;A</td>\n",
       "      <td></td>\n",
       "      <td>PMID:10874631</td>\n",
       "      <td>singlecohort/jsons/PMID_10874631_II2.json</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4290</th>\n",
       "      <td>PATIENT II.1</td>\n",
       "      <td>singlecohort</td>\n",
       "      <td>OMIM:301310</td>\n",
       "      <td>Anemia, sideroblastic, and spinocerebellar ataxia</td>\n",
       "      <td>ABCB7</td>\n",
       "      <td>NM_001271696.3:c.1231G&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>PMID:11118249</td>\n",
       "      <td>singlecohort/jsons/PMID_11118249_PATIENTII1.json</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            patient_id        cohort   disease_id  \\\n",
       "2210  Family 1 proband  singlecohort  OMIM:148600   \n",
       "3458  Family 3 proband  singlecohort  OMIM:148600   \n",
       "160   Family 2 proband  singlecohort  OMIM:148600   \n",
       "88                II.2  singlecohort  OMIM:601718   \n",
       "4290      PATIENT II.1  singlecohort  OMIM:301310   \n",
       "\n",
       "                                                disease   gene  \\\n",
       "2210        Keratoderma, palmoplantar, punctate type IA  AAGAB   \n",
       "3458        Keratoderma, palmoplantar, punctate type IA  AAGAB   \n",
       "160         Keratoderma, palmoplantar, punctate type IA  AAGAB   \n",
       "88                              Retinitis pigmentosa 19  ABCA4   \n",
       "4290  Anemia, sideroblastic, and spinocerebellar ataxia  ABCB7   \n",
       "\n",
       "                      allele_1 allele_2           PMID  \\\n",
       "2210  NM_024666.5:c.505_506dup           PMID:28239884   \n",
       "3458    NM_024666.5:c.870+1G>A           PMID:28239884   \n",
       "160       NM_024666.5:c.473del           PMID:28239884   \n",
       "88     NM_000350.3:c.1938-1G>A           PMID:10874631   \n",
       "4290  NM_001271696.3:c.1231G>C           PMID:11118249   \n",
       "\n",
       "                                                  filename  \n",
       "2210  singlecohort/jsons/PMID_28239884_Family1proband.json  \n",
       "3458  singlecohort/jsons/PMID_28239884_Family3proband.json  \n",
       "160   singlecohort/jsons/PMID_28239884_Family2proband.json  \n",
       "88               singlecohort/jsons/PMID_10874631_II2.json  \n",
       "4290      singlecohort/jsons/PMID_11118249_PATIENTII1.json  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = stats.get_summary_df().sort_values(by='gene')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Individual statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_in_days</th>\n",
       "      <th>age_in_years</th>\n",
       "      <th>vital_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMID_34722527_individual_048-051_1_Thaddeus_P__Dryja_Null RPGRIP1 Al-individual_048-051_1_Thaddeus_P__Dryja_Null RPGRIP1 Al</td>\n",
       "      <td>UNKNOWN_SEX</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMID_23407777_23407777_P1-23407777_P1</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>44.00</td>\n",
       "      <td>0.120465</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMID_31239556_individual_22_father-individual 22, father</td>\n",
       "      <td>MALE</td>\n",
       "      <td>11322.75</td>\n",
       "      <td>31.000000</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMID_29469822_Family_4_II-2-Family 4 II-2</td>\n",
       "      <td>MALE</td>\n",
       "      <td>4.00</td>\n",
       "      <td>0.010951</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMID_31021519_SATB2_47_from_Zarate_et_al__2018a__Bengani_et_al-SATB2-47 from Zarate et al., 2018a;  Bengani et al.</td>\n",
       "      <td>MALE</td>\n",
       "      <td>2556.75</td>\n",
       "      <td>7.000000</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PMID_37196654_Individual_5-Individual 5</td>\n",
       "      <td>MALE</td>\n",
       "      <td>9131.25</td>\n",
       "      <td>25.000000</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PMID_29290338_Family_UAB_R45201FN_101_individual_RS-Family UAB-R45201FN.101 individual RS</td>\n",
       "      <td>MALE</td>\n",
       "      <td>1461.00</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>UNKNOWN_STATUS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PMID_36446582_Novara_2017_P2-Novara, 2017_P2</td>\n",
       "      <td>MALE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PMID_29122497_29122497_P8-29122497_P8</td>\n",
       "      <td>MALE</td>\n",
       "      <td>300.00</td>\n",
       "      <td>0.821355</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>STX_EG1010P-STX_EG1010P</td>\n",
       "      <td>UNKNOWN_SEX</td>\n",
       "      <td>1461.00</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                            id  \\\n",
       "0  PMID_34722527_individual_048-051_1_Thaddeus_P__Dryja_Null RPGRIP1 Al-individual_048-051_1_Thaddeus_P__Dryja_Null RPGRIP1 Al   \n",
       "1                                                                                        PMID_23407777_23407777_P1-23407777_P1   \n",
       "2                                                                     PMID_31239556_individual_22_father-individual 22, father   \n",
       "3                                                                                    PMID_29469822_Family_4_II-2-Family 4 II-2   \n",
       "4           PMID_31021519_SATB2_47_from_Zarate_et_al__2018a__Bengani_et_al-SATB2-47 from Zarate et al., 2018a;  Bengani et al.   \n",
       "5                                                                                      PMID_37196654_Individual_5-Individual 5   \n",
       "6                                    PMID_29290338_Family_UAB_R45201FN_101_individual_RS-Family UAB-R45201FN.101 individual RS   \n",
       "7                                                                                 PMID_36446582_Novara_2017_P2-Novara, 2017_P2   \n",
       "8                                                                                        PMID_29122497_29122497_P8-29122497_P8   \n",
       "9                                                                                                      STX_EG1010P-STX_EG1010P   \n",
       "\n",
       "           sex  age_in_days  age_in_years    vital_status  \n",
       "0  UNKNOWN_SEX          NaN           NaN            None  \n",
       "1       FEMALE        44.00      0.120465            None  \n",
       "2         MALE     11322.75     31.000000            None  \n",
       "3         MALE         4.00      0.010951            None  \n",
       "4         MALE      2556.75      7.000000            None  \n",
       "5         MALE      9131.25     25.000000            None  \n",
       "6         MALE      1461.00      4.000000  UNKNOWN_STATUS  \n",
       "7         MALE          NaN           NaN            None  \n",
       "8         MALE       300.00      0.821355            None  \n",
       "9  UNKNOWN_SEX      1461.00      4.000000            None  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from ppktstore.release.stats import summarize_individuals\n",
    "\n",
    "individuals_df = summarize_individuals(store)\n",
    "individuals_df.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summary statistics\n",
    "\n",
    "\n",
    "#### Sex\n",
    "The number of males and females in all case report collections."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'males': 1848, 'females': 1611, 'unknown': 1507}"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sex_summary = {\n",
    "    'males': sum(individuals_df.sex=='MALE'),\n",
    "    'females': sum(individuals_df.sex=='FEMALE'),\n",
    "    'unknown': sum(individuals_df.sex=='UNKNOWN_SEX')\n",
    "}\n",
    "sex_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3459 (69.7%) had the sex specified (53.4% males, 46.6% females)'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_w_sex = sex_summary['males'] + sex_summary['females']\n",
    "perc_w_sex = (100 * n_w_sex) / sum(sex_summary.values())\n",
    "perc_males = (100 * sex_summary['males']) / n_w_sex\n",
    "perc_females = (100 * sex_summary['females']) / n_w_sex\n",
    "\n",
    "f'{n_w_sex} ({perc_w_sex:.1f}%) had the sex specified ({perc_males:.1f}% males, {perc_females:.1f}% females)'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "#### Age\n",
    "The number and percentage of subjects with the age information available."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'individuals with no age': '1977 (39.8%)',\n",
       " 'individuals with age': '2989 (60.2%)'}"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_no_age = sum(individuals_df.age_in_days.isna())\n",
    "n_w_age = len(individuals_df) - n_no_age\n",
    "age_summary = {\n",
    "    'individuals with no age': f'{n_no_age} ({n_no_age * 100 / len(individuals_df):.1f}%)',\n",
    "    'individuals with age': f'{n_w_age} ({n_w_age * 100 / len(individuals_df):.1f}%)',\n",
    "}\n",
    "age_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>item</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>version</td>\n",
       "      <td>0.1.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>phenopackets</td>\n",
       "      <td>4966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>diseases</td>\n",
       "      <td>378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>genes</td>\n",
       "      <td>343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>alleles</td>\n",
       "      <td>2934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PMIDs</td>\n",
       "      <td>729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>individuals per gene (max)</td>\n",
       "      <td>459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>individuals per gene (min)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>individuals per gene (mean)</td>\n",
       "      <td>14.478134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>individuals per gene (median)</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>individuals per disease (max)</td>\n",
       "      <td>459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>individuals per disease (min)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>individuals per disease (mean)</td>\n",
       "      <td>13.137566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>individuals per disease (median)</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>genes associated with a single disease</td>\n",
       "      <td>319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>genes associated with two diseases</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>genes associated with multiple diseases</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>gene(s) with maximum number of diseases</td>\n",
       "      <td>FBN1 with 5 associated diseases</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>total HPO terms used</td>\n",
       "      <td>69507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>present HPO terms used</td>\n",
       "      <td>37081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>excluded HPO terms used</td>\n",
       "      <td>32426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>unique HPO terms used</td>\n",
       "      <td>2618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>present HPO terms per cohort (mean)</td>\n",
       "      <td>37081.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>present HPO terms per cohort (median)</td>\n",
       "      <td>37081.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>present HPO terms per cohort (min)</td>\n",
       "      <td>37081.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>present HPO terms per cohort (max)</td>\n",
       "      <td>37081.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>excluded HPO terms per cohort (mean)</td>\n",
       "      <td>32426.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>excluded HPO terms per cohort (median)</td>\n",
       "      <td>32426.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>excluded HPO terms per cohort (min)</td>\n",
       "      <td>32426.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>excluded HPO terms per cohort (max)</td>\n",
       "      <td>32426.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>total HPO terms per cohort (mean)</td>\n",
       "      <td>69507.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>total HPO terms per cohort (median)</td>\n",
       "      <td>69507.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>total HPO terms per cohort (min)</td>\n",
       "      <td>69507.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>total HPO terms per cohort (max)</td>\n",
       "      <td>69507.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       item                            value\n",
       "0                                   version                           0.1.19\n",
       "1                              phenopackets                             4966\n",
       "2                                  diseases                              378\n",
       "3                                     genes                              343\n",
       "4                                   alleles                             2934\n",
       "5                                     PMIDs                              729\n",
       "6                individuals per gene (max)                              459\n",
       "7                individuals per gene (min)                                1\n",
       "8               individuals per gene (mean)                        14.478134\n",
       "9             individuals per gene (median)                              3.0\n",
       "10            individuals per disease (max)                              459\n",
       "11            individuals per disease (min)                                1\n",
       "12           individuals per disease (mean)                        13.137566\n",
       "13         individuals per disease (median)                              4.0\n",
       "14   genes associated with a single disease                              319\n",
       "15       genes associated with two diseases                               15\n",
       "16  genes associated with multiple diseases                                9\n",
       "17  gene(s) with maximum number of diseases  FBN1 with 5 associated diseases\n",
       "18                     total HPO terms used                            69507\n",
       "19                   present HPO terms used                            37081\n",
       "20                  excluded HPO terms used                            32426\n",
       "21                    unique HPO terms used                             2618\n",
       "22      present HPO terms per cohort (mean)                          37081.0\n",
       "23    present HPO terms per cohort (median)                          37081.0\n",
       "24       present HPO terms per cohort (min)                          37081.0\n",
       "25       present HPO terms per cohort (max)                          37081.0\n",
       "26     excluded HPO terms per cohort (mean)                          32426.0\n",
       "27   excluded HPO terms per cohort (median)                          32426.0\n",
       "28      excluded HPO terms per cohort (min)                          32426.0\n",
       "29      excluded HPO terms per cohort (max)                          32426.0\n",
       "30        total HPO terms per cohort (mean)                          69507.0\n",
       "31      total HPO terms per cohort (median)                          69507.0\n",
       "32         total HPO terms per cohort (min)                          69507.0\n",
       "33         total HPO terms per cohort (max)                          69507.0"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_d = stats.get_descriptive_stats(version=release_tag)\n",
    "items = list()\n",
    "for k,v in stats_d.items():\n",
    "    items.append({\"item\": k, \"value\": v})\n",
    "pd.DataFrame(items)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
